File: 프로그래머스/3/161988. 연속 펄스 부분 수열의 합/연속 펄스 부분 수열의 합.js
// 결국 연속 수열 * 펄스수열
// slidingwindow?
function solution(sequence) {
    var dp = [];
    dp.push(sequence[0]);
    for(var i=1; i<sequence.length; i++){
        var sign = i%2===0?1:-1;
        dp[i]=dp[i-1]+(sequence[i]*sign);
    }
    dp.sort((a,b)=>a-b);
    var answer = Math.max(Math.abs(dp[0]-dp[dp.length-1]),Math.abs(dp[0]),Math.abs(dp[dp.length-1]));
    
    return answer;
}

File: 프로그래머스/3/161988. 연속 펄스 부분 수열의 합/연속 펄스 부분 수열의 합.java
import java.util.*;
class Solution {
    public long solution(int[] sequence) {
        final int len = sequence.length; 
        long[] acc1 = new long[len];
        acc1[0] = sequence[0];
        for(int i=1; i<len; i++){
            acc1[i]=i%2==0?acc1[i-1]+sequence[i]:acc1[i-1]+sequence[i]*(-1);
        }
        Arrays.sort(acc1);
        long answer = 0;
        answer = Math.max(Math.abs(acc1[0]),Math.abs(acc1[len-1]));
        answer = Math.max(answer,Math.abs(acc1[0]-acc1[len-1]));
        return answer;
    }
}